### PR TITLE
Fix macro scoping

### PR DESCRIPTION
**Pull request by musm**
Addressed @yuyichao suggestion for macro scoping.

### CODE REVIEW

**Comment by musm** [comment]
rebased

File: src/mxarray.jl [hunk 1]
```diff
@@ -173,55 +173,55 @@ const _mx_is_empty   = mxfunc(:mxIsEmpty)
 const _mx_is_struct  = mxfunc(:mxIsStruct)
 const _mx_is_cell    = mxfunc(:mxIsCell) 
 
-macro mxget_attr(fun, ret)
-    :(ccall($(fun)::Ptr{Void}, $(ret), (Ptr{Void},), mx))
+macro mxget_attr(fun, ret, mx)
+    :(ccall($(fun)::Ptr{Void}, $(ret), (Ptr{Void},), $(mx)))
```
Comment (yuyichao) [inline src/mxarray.jl]: With the bug fixed this won't work anymore. The correct way to write this is

```jl
:(ccall($(esc(fun))::Ptr{Void}, $(esc(ret)), (Ptr{Void},), $(esc(mx))))
```

**Comment by musm** [comment]
done

**Comment by musm** [comment]
done & tests pass locally

File: src/mxarray.jl [hunk 2]
```diff
@@ -173,55 +173,55 @@ const _mx_is_empty   = mxfunc(:mxIsEmpty)
 const _mx_is_struct  = mxfunc(:mxIsStruct)
 const _mx_is_cell    = mxfunc(:mxIsCell) 
 
-macro mxget_attr(fun, ret)
-    :(ccall($(fun)::Ptr{Void}, $(ret), (Ptr{Void},), mx))
+macro mxget_attr(fun, ret, mx)
+    :(ccall($(esc(fun))::Ptr{Void}, $(esc(ret)), (Ptr{Void},), $(esc(mx))))
```
Comment (musm) [inline src/mxarray.jl]: I feel like the type assertion is kinda verbose/unecessary but w/e

Comment (yuyichao) [inline src/mxarray.jl]: The type assertion is not necessary although the use is also not precompilable. The right way is to use a global const `Ref` and initialize it in `__init__`.

Comment (musm) [inline src/mxarray.jl]: Which object would be `Ref` ?

Comment (yuyichao) [inline src/mxarray.jl]: The const globals.

**Pull request merged**